### PR TITLE
jsonrpc,rpc/client: add accountunlocked

### PR DESCRIPTION
**Pull request by chappjc**
This adds the `accountunlocked` command to the JSON-RPC server. It is similar to the gRPC accountunlocked command, but the result also indicates if the account is even encrypted (lockable).

Should be backported to release-v1.6, along with https://github.com/decred/dcrwallet/pull/1969 probably.

### CODE REVIEW

**Comment by chappjc** [comment]
These changes, including the added rpc/client methods, are tested in https://github.com/decred/dcrdex/pull/1023.

Also tested with dcrctl built from its release-v1.6 with `replace decred.org/dcrwallet => github.com/chappjc/dcrwallet v1.6.3-0.20210324151833-873c564976fd` (port of this PR to dcrwallet release-v1.6).

File: internal/rpchelp/helpdescs_en_US.go [hunk 1]
```diff
@@ -24,6 +24,14 @@ var helpDescsEnUS = map[string]string{
 	"accountsyncaddressindex-branch":    "Number for the branch (0=external, 1=internal)",
 	"accountsyncaddressindex-index":     "The address index to synchronize to",
 
+	// AccountUnlockedCmd help.
+	"accountunlocked--synopsis": "Report account encryption and locked status",
+	"accountunlocked-account":   "String for the account",
```
Comment (jrick) [inline internal/rpchelp/helpdescs_en_US.go:29]: Use name, not string.  Either "Name of the account" or "Account name" would be fine here.

File: internal/rpchelp/helpdescs_en_US.go [hunk 2]
```diff
@@ -24,6 +24,14 @@ var helpDescsEnUS = map[string]string{
...
+
+	// AccountUnlockedResult help.
+	"accountunlockedresult-encrypted": "If the account is encrypted (can be individually unlocked)",
```
Comment (jrick) [inline internal/rpchelp/helpdescs_en_US.go:32]: All accounts are encrypted, so let's say "Whether the account is individually encrypted with a separate passphrase".

File: .gitignore [hunk 3]
```diff
@@ -1,6 +1,8 @@
 cmd/movefunds/movefunds
 cmd/sweepaccount/sweepaccount
-dcrwallet
+cmd/repaircfilters/repaircfilters
+cmd/treasurykey/treasurykey
+/dcrwallet
```
Comment (jrick) [inline .gitignore:5]: Can this be broken out into a different PR?

**Review by jrick — CHANGES_REQUESTED** [review]
Looks good overall but have some suggestions

File: internal/rpchelp/helpdescs_en_US.go (hunk 1 above)
Comment (chappjc) [reply internal/rpchelp/helpdescs_en_US.go:29]: Agreed. Was following the pattern with accountaddressindex-account, accountsyncaddressindex-account, but it did seem weird.

File: .gitignore (hunk 3 above)
Comment (chappjc) [reply .gitignore:5]: Sure thing.  The cmd ignores are not relevant, but git was spamming me because of the rpc/client/dcrwallet paths changed in this pr were matching the dcrwallet glob.

Comment (jrick) [reply .gitignore:5]: Oh. Yeah let's fix that then.  I wasn't sure what the reasoning was here.

**Review by jrick — APPROVED** [review]

**Comment by jrick** [comment]
Looks like you wanted this to be merged as two separate commits, can you force push with the fixups?
